### PR TITLE
Fix Seq2SeqTrainer generation path for decoder-only models

### PR DESCRIPTION
**Pull request by shaealh**
Closes #44593

## Summary
- use generation_input_ids/generation_attention_mask when provided for decoder-only models
- otherwise infer prompt from leading -100 labels and build left-padded prompt batch
- return completion tokens for decoder-only generation (strip prompt)
- keep encoder-decoder behavior unchanged

## Tests
- PYTHONPATH=src python -m pytest tests/trainer/test_trainer_seq2seq.py -k Seq2SeqTrainerPredictionStepTester -q -rs

Related #26474, #33396
Follow-up to #32346

### CODE REVIEW

**Comment by Rocketknight1** [comment]
cc @sunmarc

**Comment by shaealh** [comment]
Still waiting for review and approval

**Comment by shaealh** [comment]
following up here. Still waiting for approval
